sonic-xcvrd/xcvrd/dom/dom_mgr.py

```diff
@@ -210,17 +258,16 @@ def task_worker(self):
         # Set the periodic db update time
         dom_info_update_periodic_secs = self.DOM_INFO_UPDATE_PERIOD_SECS
 
-        # Adding dom_info_update_periodic_secs to allow xcvrd to initialize ports
-        # before starting the periodic update
-        next_periodic_db_update_time = datetime.datetime.now() + datetime.timedelta(seconds=dom_info_update_periodic_secs)
-        is_periodic_db_update_needed = False
+        # Wait for all PORTs to be initialized
+        remaining_ports = self.wait_port_initialization(self.DOM_WAIT_PORT_INIT_TIMEOUT_SECS)
+        if remaining_ports:
+            self.log_error("wait_port_initialization() timed out after {} seconds, remaining ports: {}".format(
+                    self.DOM_WAIT_PORT_INIT_TIMEOUT_SECS, remaining_ports))
+        else:
+            self.log_notice("All ports are in CMIS terminal state, start DOM monitoring")
 
         # Start loop to update dom info in DB periodically and handle port change events
```

Comment:
@prgeor Please add `is_periodic_db_update_needed = True` to ensure that DOM data is updated periodically